js/rook.js

```diff
@@ -0,0 +1,86 @@
+import {Piece} from "./piece.js";
+
+export class Rook extends Piece {
+    /**
+     * Constructor: inherits from abstract Piece class
+     * @param id piece id
+     * @param rowPos current row position
+     * @param colPos current column position
+     * @param playerId player id
+     */
+    constructor(id, rowPos, colPos, playerId) {
+        super();
+        this.id = id;
+        this.rowPos = rowPos;
+        this.colPos = colPos;
+        this.playerId = playerId;
+        this.unicodeChar = "&#9814;";
+    }
+
+    /**
+     * Moves piece to {row}, {col} and updates board
+     * @param row new row
+     * @param col new column
+     * @param board Board.board
+     */
+    movePiece(row, col, board) {
+        let possibleMoves = this.possibleMoves(board);
+        if (!this.validateMove(row, col, possibleMoves)) { // ignore invalid move
+            return;
+        }
+
+        if (board[row][col]) { // taking a piece
+            this.takenPiece = board[row][col];
+            this.takenPiece.setTaken();
+            board[row][col] = null;
+        }
+
+        // Update board
+        board[this.rowPos][this.colPos] = null;
+        board[row][col] = this;
+
+        // Update piece position
+        this.rowPos = row;
+        this.colPos = col;
+    }
+
+    /**
+     * Gets a list of possible moves the rook piece can make including attack moves
+     * @param board Board singleton instance
+     * @returns {*[]}
+     */
+    possibleMoves(board) {
+        let possibleMoves = [];
+        for (let i = this.rowPos + 1; i < 8; i++) { // check below
+            if (board[i][this.colPos]) {
+                possibleMoves.push([i, this.colPos]);
+                break;
+            }
```

Comment:
Don't forget to verify that the piece that's being taken doesn't belong to the current player.

Reply:
added, thanks!